server/controllers/authentication.js

```diff
@@ -0,0 +1,35 @@
+require('dotenv').config({ silent: true });
+const jwt = require('jsonwebtoken');
+
+/**
+ * class for JWT verification and authorization
+ */
+class Authentication {
+
+  /**
+   * method verify to verify if the token is valid
+   * @param {object} req - request
+   * @param {object} res - response
+   * @param {function} next function
+   * @return {boolean} true if token is valid;
+   */
+  static verify(req, res, next) {
+    if (req.headers.authorization) {
+      jwt.verify(req.headers.authorization, process.env.SECRET,
+        (err, decoded) => {
+          if (err) {
+            return res.status(401).json({
+              done: false,
+              message: 'Token authentication failed' });
+          }
+          req.decoded = decoded;
+          req.token = req.headers.authorization;
+          next();
+        });
+    } else {
+      return res.status(401).send({ done: false, message: 'Please login' });
+    }
```

Comment:
Not much of a big deal, but normally following a defensive approach is much more easier to read, you will first check if the headers do not contain the authorization token and return early if not, that way you just have an if block without necessarily the else part and an easier to ready code.